### PR TITLE
🏗 Apply `rtv_version` flag to dist and build tasks

### PR DESCRIPTION
**Pull request by micajuine-ho**
Changed the internalRuntimeVersion's existing `version` flag to `rtv_version` to avoid triggering `node/gulp --version`.

Added `rtv_version` to list of flags for `dist` and `build`, with documentation.

### CODE REVIEW

File: build-system/tasks/build.js [hunk 1]
```diff
@@ -132,6 +132,8 @@ build.flags = {
   noextensions: '  Builds with no extensions.',
   core_runtime_only: '  Builds only the core runtime.',
   coverage: '  Adds code coverage instrumentation to JS files using istanbul.',
+  rtv_version:
+    '  Specifies the version number instead of creating a new number',
```
Comment (rsimha) [inline build-system/tasks/build.js]: For clarity, can you rewrite as follows?

```
Overrides the version number written to AMP_CONFIG
```

File: build-system/tasks/build.js [hunk 2]
```diff
@@ -132,6 +132,8 @@ build.flags = {
...
 };
 
 watch.description = 'Watches for changes in files, re-builds when detected';
```
Comment (rsimha) [inline build-system/tasks/build.js]: This flag will work with `watch` and `defaultTask` as well, so let's add them there, and to the dev docs too.

File: contributing/TESTING.md [hunk 3]
```diff
@@ -57,6 +57,7 @@ Before running these commands, make sure you have Node.js, yarn, and Gulp instal
...
 | `gulp dist --fortesting`                                  | Builds production binaries for local testing. (Allows use cases like ads, tweets, etc. to work with minified sources. Overrides `TESTING_HOST` if specified. Uses the production `AMP_CONFIG` by default.)                                    |
 | `gulp dist --fortesting --config=<config>`                | Builds production binaries for local testing, with the specified `AMP_CONFIG`. `config` can be `prod` or `canary`. (Defaults to `prod`.)                                                                                                      |
+| `gulp dist --rtv_version=<rtv_version>`                   | Builds production binaries with the specified version number.                                                                                                                                                                                 |
```
Comment (rsimha) [inline contributing/TESTING.md]: Thanks for remembering to update this file :)

File: build-system/tasks/build.js (hunk 2 above)
Comment (micajuine-ho) [reply build-system/tasks/build.js]: Good call

**Comment by micajuine-ho** [comment]
> Also, since you're already editing dist.js, could I ask you to make the following improvement to the logging? 

So now we always want to call `printConfigHelp()`?

**Comment by rsimha** [comment]
> So now we always want to call `printConfigHelp()`?

You are correct. Earlier, we were calling it only with `--fortesting`, but there are genuine reasons for developers to invoke `gulp dist` without the `--fortesting` flag.

**Comment by mdmower** [comment]
This PR does not look correct. A _version_ is not the same as _rtv version_. An _rtv version_ includes the two digit config (00 for canary, 01 for production, 02+ for experiments) in front of the _version_. As you've modified `getVersion()`, it now outputs _version_ if generated automatically or an _rtv version_ if manually specified by flag `--rtv_version`. This is inconsistent, not to mention you've likely broken build system usage for some folks who suddenly cannot use the `--version` flag.

For example, see usage of `--version` and `--config` to generate a full _rtv version_ in my WIP self hosting guide: https://gist.github.com/mdmower/b56e94f0dc36beafb825b0c5e31fceef#build-the-framework .

**Pull request merged**